src/indexing.jl

```diff
@@ -11,6 +11,24 @@ function _getindex_scalar(t, idxs)
     t.data[first(i)]
 end
 
+Base.haskey(t::NDSparse, idxs) = (flush!(t); _haskey(t, idxs))
+
+_haskey(t::NDSparse{T,D}, idxs::D) where {T,D<:Tuple} = _haskey_scalar(t, idxs)
+
+function _haskey_scalar(t, idxs)
+    i = searchsorted(t.index, convertkey(t, idxs))
+    return length(i) == 1
+end
+
+Base.get(t::NDSparse, idxs, default) = (flush!(t); _get(t, idxs, default))
```

Comment:
I'm also fond of @piever's suggestion to add a `get(function, t, idxs)` method if you're up for it.

Reply:
I added that and used it to implement `_getindex_scalar`.